<!doctype html>
<html lang="es">
<head>
<link rel="stylesheet" type="text/css" href="base.css" />
<link rel="stylesheet" type="text/css" href="exe_highlighter.css" />
<link rel="stylesheet" type="text/css" href="content.css" />
<link rel="stylesheet" type="text/css" href="nav.css" />
<meta http-equiv="content-type" content="text/html;  charset=utf-8" />
<title>2. Excepciones | UA02. Primer acceso a datos </title>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<meta name="author" content="David Bermúdez" />
<link rel="license" type="text/html" href="http://creativecommons.org/licenses/by-sa/4.0/" />
<meta name="generator" content="eXeLearning 2.7 - exelearning.net" />
<!--[if lt IE 9]><script type="text/javascript" src="exe_html5.js"></script><![endif]-->
<script type="text/javascript" src="exe_jquery.js"></script>
<script type="text/javascript" src="exe_highlighter.js"></script>
<script type="text/javascript" src="common_i18n.js"></script>
<script type="text/javascript" src="common.js"></script>
<link rel="stylesheet" type="text/css" href="udl-content.css" />
<script type="text/javascript" src="udl-content.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
	</head>
<body class="exe-web-site" id="exe-node-5"><script type="text/javascript">document.body.className+=" js"</script>
<div id="content">
<p id="skipNav"><a href="#main" class="sr-av">Saltar la navegación</a></p>
<header id="header" ><div id="headerContent">UA02. Primer acceso a datos</div></header>
<nav id="siteNav">
<ul>
   <li><a href="index.html" class="daddy main-node">UA02. Primer acceso a datos</a></li>
   <li><a href="1_ficheros_xml.html" class="daddy">1. Ficheros XML</a>
   <ul class="other-section">
      <li><a href="11_xml_con_dom.html" class="no-ch">1.1. XML con DOM</a></li>
      <li><a href="12_creacin_de_documentos_con_dom.html" class="no-ch">1.2. Creación de documentos con DOM</a></li>
      <li><a href="12_xml_con_sax.html" class="no-ch">1.2. XML con SAX</a></li>
      <li><a href="13_conversiones.html" class="no-ch">1.3. Conversiones</a></li>
   </ul>
   </li>
   <li id="active"><a href="2_excepciones.html" class="active daddy">2. Excepciones</a>
   <ul>
      <li><a href="21_excepciones_controladas.html" class="no-ch">2.1. Excepciones controladas</a></li>
      <li><a href="22_excepciones_no_controladas.html" class="no-ch">2.2. Excepciones no controladas</a></li>
   </ul>
   </li>
   <li><a href="3_pruebas_y_documentacin.html" class="no-ch">3. Pruebas y documentación</a></li>
</ul>
</nav>
<div id='topPagination'>
<nav class="pagination noprt">
<a href="13_conversiones.html" class="prev"><span><span>&laquo; </span>Anterior</span></a> <span class="sep">| </span><a href="21_excepciones_controladas.html" class="next"><span>Siguiente<span> &raquo;</span></span></a>
</nav>
</div>
<div id="main-wrapper">
<section id="main">
<header id="nodeDecoration"><h1 id="nodeTitle">2. Excepciones</h1></header>
<article class="iDevice_wrapper UDLcontentIdevice" id="id7">
<div class="iDevice emphasis0" >
<div id="ta7_839_2" class="block iDevice_content">
<div class="exe-udlContent exe-udlContent-engagement"><section class="exe-udlContent-block"><div class="exe-udlContent-content"><div class="exe-udlContent-content-main"><p>El tratamiento de las excepciones es un proceso muy importante en la creación de un programa. Es el mecanismo que nos permite controlar un proceso anómalo dentro de nuestro programa y gestionar qué ocurre cuando se produce un error.</p>
<p><strong>Una excepción es un evento que se produce durante la ejecución de un programa que interrumpe el flujo normal de ejecución del código a causa de un error, ya sea controlado o inesperado.</strong></p>
<p><img src="try-catch-flow.svg" alt="Try-catch" style="margin-right: 10px; margin-left: 10px; float: right;" width="320" height="256">Para poder gestionar los errores, en Java utilizamos lo que se llama un <strong>try-catch</strong>.</p>
<p>Un try-catch es un bloque de código que se añadirá a nuestro programa siempre que queramos controlar una parte del código que vayamos a desarrollar. El código se ejecutará en orden e irá siguiendo hasta que se produzca un error, en ese momento entrará al catch.</p>
<p>Podemos definir más de un catch en un mismo try. Cuando se produzca el error, entrará en el primer bloque catch que coincida con el tipo de excepción y solo podrá entrar en uno de los bloques catch.</p>
<p>Los tipos de excepción más específicos deben aparecer primero en la estructura, seguidos de los tipos de excepción más generales.</p>
<p>El <strong>try</strong> es lo que primero se ejecuta: se prueba la acción que se intenta realizar y, si se produce un error, el <strong>catch</strong> recoge la excepción y, según lo que indiquemos, lanza un mensaje con la descripción de lo que ha ocurrido. Al lanzar una excepción, el proceso normal del programa se interrumpirá.</p>
<p>Lo mejor para controlar los errores es tener en cuenta todas las posibilidades de error y controlarlas. Se debe utilizar un try-catch siempre que sea necesario.</p></div></div></section></div>
</div>
</div>
</article>
<article class="iDevice_wrapper UDLcontentIdevice em_iDevice em_iDevice_observe" id="id8">
<div class="iDevice emphasis1" >
<header class="iDevice_header" style="background-image:url(icon_observe.png)"><h1 class="iDeviceTitle">Ejemplo</h1></header>
<div class="iDevice_inner">
<div class="iDevice_content_wrapper">
<div id="ta8_840_2" class="block iDevice_content">
<div class="exe-udlContent exe-udlContent-engagement"><section class="exe-udlContent-block"><div class="exe-udlContent-content"><div class="exe-udlContent-content-main"><div class="highlighted-code language-java line-numbers">
<div>
<pre><code>try{
    //Parte donde se ejecuta el código que puede dar excecion
    File prueba = new File("c\\:prueba.txt");
} catch( NoSuchFileException e){
    System.out.println(“No se ha encontrado el archivo” +e);
} catch(IOExcepction ex){
    //Parte donde se gestiona la excepción
    throw ex;
} finally {
    //Bloque de código que se ejecutará siempre, aunque se lanze una //excepcion
}</code></pre>
</div>
</div>
<p>Como vemos en el ejemplo, antes de realizar la acción debemos envolverla con un <strong>try-catch</strong>.</p>
<p>Dentro del bloque del <strong>try</strong>, añadiremos la parte del código que queremos controlar si se produce un error. Si no se produce el error, el programa continuará la ejecución y entrará dentro del bloque <strong>finally</strong>. Este bloque se ejecutará siempre, aunque no se produzca un error, y es especialmente útil para cerrar el proceso o ejecutar algún tipo de acción necesaria. En el caso de que se produzca un error, entrará en el <strong>catch</strong> y realizará las acciones dentro de ese apartado y lanzará un error.</p></div></div></section></div>
</div>
</div>
</div>
</div>
</article>
<div id="packageLicense" class="cc cc-by-sa">
<p><span>Obra publicada con</span> <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Licencia Creative Commons Reconocimiento Compartir igual 4.0</a></p>
</div>
</section>
</div>
<div id='bottomPagination'>
<nav class="pagination noprt">
<a href="13_conversiones.html" class="prev"><span><span>&laquo; </span>Anterior</span></a> <span class="sep">| </span><a href="21_excepciones_controladas.html" class="next"><span>Siguiente<span> &raquo;</span></span></a>
</nav>
</div>
</div>
<script type="text/javascript" src="_cedec_js.js"></script></body></html>